docs/PRO_DEF_WEAK_DISPATCH.md

```diff
@@ -26,7 +26,9 @@ struct dispatch_name : existing_dispatch {
 
 ## Notes
 
-A weak dispatch can extend an existing dispatch with a default implementation. This is useful when instantiating a `proxy<F>` with a value that does not support some conventions defined by `F`. Similar with [`PRO_DEF_FREE_DISPATCH`](PRO_DEF_FREE_DISPATCH.md), `default_func_name` can be the name of an arbitrary function or anything that supports `()` syntax, including a constructor.
+A "weak dispatch" can extend an existing dispatch with a default implementation. This is useful when instantiating a `proxy<F>` with a value that does not support some conventions defined by `F`. Similar to [`PRO_DEF_FREE_DISPATCH`](PRO_DEF_FREE_DISPATCH.md), `default_func_name` can be the name of an arbitrary function or anything that supports `()` syntax, including a constructor.
+
+Unlike a "default method" in [Java](https://docs.oracle.com/javase/specs/jls/se23/html/jls-9.html#jls-9.4-200) or [C#](https://learn.microsoft.com/dotnet/csharp/language-reference/proposals/csharp-8.0/default-interface-methods), a "weak dispatch" in the "Proxy" library is only intended to improve compilation speed and binary size when the contained value of a `proxy` object does not participate code generation. If a "default implementation" otherwise needs to observe the contained value of a `proxy` object, it is encouraged to define a separate free function, and subsequently define a dispatch type of it by using [`PRO_DEF_FREE_DISPATCH`](PRO_DEF_FREE_DISPATCH.md) or [`PRO_DEF_FREE_AS_MEM_DISPATCH`](PRO_DEF_FREE_AS_MEM_DISPATCH.md).
```

Comment:
I think "improve compilation speed and binary size when the contained value of a `proxy` object does not participate code generation" is somehow a viewport from the library authors, but for the users, it is not clear enough when they will want to use this.

To me, it feels that the weak dispatch is just for satisfying the requirements of a facade when some non essential methods are not always implemented.  And  the weak dispatch should generally not be expected to be called normally (e.g in the provided samples, the weak dispatches just throw something like "Not implemented" or return -1 for hash).  But I am not sure how it will compare with one base facade with only the essential methods, and another facade first `add_facade` the base facade and adds some additional methods.

Reply:
Updated the comparison part per your comments. Does it look better?

Reply:
looks good to me.